worker/provisioner/provisioner_task.go

```diff
@@ -375,6 +375,11 @@ func (task *provisionerTask) processProfileChanges(ids []string) error {
 		}
 		m := mResult.Machine
 		removeDoc, err := processOneMachineProfileChange(m, profileBroker)
+		// The machine is not provisioned yet, so we should continue on
+		// and apply the information at a later stage.
```

Comment:
Let's be more specific.  :-) "The machine is not provisioned yet, therefore we can continue and the profile will be applied when the machine is provisioned."